### PR TITLE
 Packaging: remove setup.cfg, move info into pyproject.toml

### PR DESCRIPTION
**Pull request by GenevieveBuckley**
Closes https://github.com/napari/cookiecutter-napari-plugin/issues/167

> Now that [pyproject.toml supports project metadata](https://peps.python.org/pep-0621/) (classifiers, author info, etc.) we should remove the [setup.cfg](https://github.com/napari/cookiecutter-napari-plugin/blob/main/%7B%7Bcookiecutter.plugin_name%7D%7D/setup.cfg) file and put the information into [pyproject.toml](https://github.com/napari/cookiecutter-napari-plugin/blob/main/%7B%7Bcookiecutter.plugin_name%7D%7D/pyproject.toml)
> 
> Advantages:
> * simplifies the repository
> * It makes it much easier for users to retro-fit napari plugins to existing
> * it discourages people from accidentally duplicating info across both pyproject.toml and setup.cfg (if this happens pyproject.toml overwrites the information).  
>        
>    For an in the wild example of this problem, Draga and I got an email from some users confused about why their plugin was not being picked up by the napari hub. It turned out that they had classifier tags in both pyproject.toml AND setup..cfg, and so the napari framework tag was not being included in the packaged build (and therefore not available on PyPI for the napari-hub to discover it)

### CODE REVIEW

**Comment by Czaki** [comment]
Did you need help with debug, or just do not have time?

**Comment by GenevieveBuckley** [comment]
This is ready for review

File: {{cookiecutter.plugin_name}}/tox.ini [hunk 1]
```diff
@@ -1,13 +1,14 @@
 # For more information about tox, see https://tox.readthedocs.io/en/latest/
 [tox]
-envlist = py{38,39,310}-{linux,macos,windows}
+envlist = py{38,39,310,311}-{linux,macos,windows}
```
Comment (Czaki) [inline {{cookiecutter.plugin_name}}/tox.ini:3]:
```suggestion
envlist = py{39,310,311,3.12}-{linux,macos,windows}
```

File: {{cookiecutter.plugin_name}}/tox.ini [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
 isolated_build=true
 
 [gh-actions]
 python =
     3.8: py38
     3.9: py39
     3.10: py310
+    3.11: py311
```
Comment (Czaki) [inline {{cookiecutter.plugin_name}}/tox.ini:11]:
```suggestion
    3.9: py39
    3.10: py310
    3.11: py311 
    3.12: py312
```

File: tox.ini [hunk 3]
```diff
@@ -1,11 +1,12 @@
 [tox]
 recreate = true
 skipsdist = true
-envlist = py{38,39,310}
+envlist = py{38,39,310,311}
```
Comment (Czaki) [inline tox.ini:4]:
```suggestion
envlist = py{39,310,311,312}
```

File: {{cookiecutter.plugin_name}}/pyproject.toml [hunk 4]
```diff
@@ -1,3 +1,68 @@
...
+    {%- elif cookiecutter.license == "Mozilla Public License 2.0" -%}
+    "License :: OSI Approved :: Mozilla Public License 2.0 (MPL 2.0)",
+    {%- endif %}
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
```
Comment (Czaki) [inline {{cookiecutter.plugin_name}}/pyproject.toml:34]:
```suggestion
    "Programming Language :: Python :: 3.9",
    "Programming Language :: Python :: 3.10",
    "Programming Language :: Python :: 3.11",
    "Programming Language :: Python :: 3.12",
```

File: {{cookiecutter.plugin_name}}/pyproject.toml [hunk 5]
```diff
@@ -1,3 +1,68 @@
...
+    "Topic :: Scientific/Engineering :: Image Processing",
+]
+requires-python = ">=3.8"
```
Comment (Czaki) [inline {{cookiecutter.plugin_name}}/pyproject.toml:37]:
```suggestion
requires-python = ">=3.9"
```

**Review by Czaki — APPROVED** [review]
As we dropped python 3.9 already, we may want to not suggest python 3.8 compatibility for plugins.

File: {{cookiecutter.plugin_name}}/tox.ini [hunk 6]
```diff
@@ -1,13 +1,14 @@
 # For more information about tox, see https://tox.readthedocs.io/en/latest/
 [tox]
-envlist = py{38,39,310}-{linux,macos,windows}
+envlist = py{39,310,311,3.12}-{linux,macos,windows}
```
Comment (GenevieveBuckley) [inline {{cookiecutter.plugin_name}}/tox.ini:3]:
```suggestion
envlist = py{39,310,311,312}-{linux,macos,windows}
```

File: {{cookiecutter.plugin_name}}/tox.ini [hunk 7]
```diff
@@ -1,13 +1,14 @@
...
-envlist = py{38,39,310}-{linux,macos,windows}
+envlist = py{39,310,311,312}-{linux,macos,windows}
 isolated_build=true
 
 [gh-actions]
 python =
-    3.8: py38
     3.9: py39
     3.10: py310
+    3.11: py311
```
Comment (GenevieveBuckley) [inline {{cookiecutter.plugin_name}}/tox.ini:10]:
```suggestion
    3.11: py311
```

**Review by Czaki — APPROVED** [review]

**Pull request merged**